onnxruntime/core/graph/graph_utils.cc

```diff
@@ -656,9 +656,14 @@ void FinalizeNodeFusion(Graph& graph, Node& first_node, Node& second_node) {
   graph.RemoveNode(second_node.Index());
 }
 
-void FinalizeNodeFusion(Graph& graph, const std::vector<std::reference_wrapper<Node>>& nodes, Node& replacement_node) {
-  MoveAllNodeInputEdges(graph, nodes.front(), replacement_node);
-  MoveAllNodeOutputs(graph, nodes.back(), replacement_node);
+void FinalizeNodeFusion(Graph& graph, const std::vector<std::reference_wrapper<Node>>& nodes, 
+                        Node& replacement_node, bool move_input_edges, bool move_output_edges) {
+  if (move_input_edges) {
```

Comment:
>move_input_edges [](start = 6, length = 16)

I originally had a similar change to get around an issue with BiasGeluFusion. However, @askhade pointed out that this function can be called from rewrite rule implementations too and for those, Graph::Resolve() won't necessarily be called right afterward to fix up the edges. So I think if possible this function should leave the graph in a valid state (not requiring a call to Graph::Resolve()).

Reply:
Tried. not working... 
will sync with Ashwini on this 

---
In reply to: [441015357](https://github.com/microsoft/onnxruntime/pull/4167#discussion_r441015357) [](ancestors = 441015357)

Reply:
reverted this change. 

---
In reply to: [443762252](https://github.com/microsoft/onnxruntime/pull/4167#discussion_r443762252) [](ancestors = 443762252,441015357)